File: java-app/src/main/java/com/mercadolibre/java_app/models/shipping/NormalShippingOption.java
package com.mercadolibre.java_app.models.shipping;

import android.os.Parcel;
import android.os.Parcelable;

import androidx.annotation.NonNull;

import com.mercadolibre.java_app.R;
import com.mercadolibre.java_app.commons.Prices;
import com.mercadolibre.java_app.commons.ShippingOptionView;

public class NormalShippingOption extends AbstractShippingOption {
    public NormalShippingOption(Double price, String type) {
        super(price, type);
    }

    public NormalShippingOption(Double price) {
        this(price, "Normal a domicilio");
    }

    @Override
    public void showShippingPrices(@NonNull ShippingOptionView view) {
        view.showShippingPrice(Prices.priceFormat(price), R.color.black);
        view.showShippingType(type);
    }

    // -- Parcelable
    protected NormalShippingOption(Parcel in) {
        super(in);
    }

    public static final Parcelable.Creator<NormalShippingOption> CREATOR = new Parcelable.Creator<NormalShippingOption>() {
        public NormalShippingOption createFromParcel(Parcel in) {
            return new NormalShippingOption (in);
        }

        public NormalShippingOption [] newArray(int size) {
            return new NormalShippingOption[size];
        }
    };
}

File: kt-app/src/main/java/com/mercadolibre/melikotlinmeetup/models/shipping/NormalShippingOption.kt
package com.mercadolibre.melikotlinmeetup.models.shipping

import com.mercadolibre.melikotlinmeetup.R
import com.mercadolibre.melikotlinmeetup.commons.ShippingOptionView
import com.mercadolibre.melikotlinmeetup.commons.toPriceFormat
import kotlinx.android.parcel.Parcelize

@Parcelize
class NormalShippingOption(override val price: Double, override val type: String = "Normal a domicilio"): AbstractShippingOption(price, type) {
    override fun showShippingPrices(view: ShippingOptionView) {
        view.run {
            showShippingPrice(price.toPriceFormat(), R.color.black)
            showShippingType(type)
        }
    }
}